caas/kubernetes/provider/k8s.go

```diff
@@ -719,11 +729,15 @@ func (k *kubernetesClient) DeleteOperator(appName string) (err error) {
 		return errors.Trace(err)
 	}
 
+	deploymentName := appName
+	if legacy {
+		deploymentName = "juju-" + appName
```

Comment:
Considering this is repeated a few times, do you think we should have a function for this `func legacyName(name string) string`, so we don't end up fat fingering this in the future?